js/files.js

```diff
@@ -11,109 +11,114 @@
         var currentFile = $(event.target).closest('tr');
-        var selectedActions = '.selectedActions .menu-center li';
-        currentFile.find('.action-menu').click();
-        $('.filesSelectMenu').css('visibility', 'hidden');
-        $('.actions-selected').click();
-
-        var menu = currentFile.find('.fileActionsMenu');
-        var menuStyle = $('style.rightClickStyle');
-        var generateNewOption = function (action, icon, text, onClick, prepend, subOptions) {
-            return new RightClick.Option(action, text, 'icon-' + icon, typeof onClick === 'function' ? function (event, context) {
-                event.stopPropagation();
-                event.preventDefault();
-
-                menu.remove();
-                currentFile.removeClass('mouseOver');
-                currentFile.removeClass('highlighted');
-                currentFile.find('.action-menu').removeClass('open');
-
-                onClick(event, context);
-            } : onClick, subOptions);
-        };
-        var addNewOption = function (action, icon, text, onClick, prepend, subOptions) {
-            if (prepend === undefined)
-                prepend = true;
-
-            var option = generateNewOption(action, icon, text, onClick, prepend, subOptions);
-
-            if (prepend)
-                options.prepend(option);
-            else
-                options.append(option);
-        };
-        var addNewOpenSubOption = function (action, icon, text, onClick, prepend, subOptions) {
-            if (prepend === undefined)
-                prepend = true;
-
-            var option = generateNewOption(action, icon, text, onClick, prepend, subOptions);
-
-            if (prepend)
-                openSubOptions.prepend(option);
-            else
-                openSubOptions.append(option);
-        };
-
-        menu.css('visibility', 'hidden');
-
-        if (currentFile.hasClass('selected')) {
-            menu.find('ul').html('');
-
-            addNewOption('Check', 'category-disabled', t(appName, 'Unselect'), function () {
-                $(currentFile.find('input.selectCheckBox')).click();
-            });
-
-            $.each($('.selectedActions .menu-center li'), function (i, selectedAction) {
-                var action = $(selectedAction);
-
-                addNewOption(action.attr('class'), $(action.find('span.icon')).attr('class').replace('icon', '').replace(' ', '').replace('icon-', ''), $(action.find('span:not(.icon)')).text(), function () {
-                    action.find('a').click();
-                }, false);
-            });
-        }
-        else {
-            var mimeType = currentFile.attr('data-mime');
-            var text = '';
-            var icon = 'toggle';
-            var onClick = function () {
-                currentFile.find('.filename .nametext').click();
+        
+        // W2G2 Integration
+        if ($(currentFile).find(".statelock").length == 0) {
```

Comment:
You made such a big refractor, I think it would be better to only check if the file is locked and if it is, just return null so the menu does not even process :)

Reply:
I think the spaces made the big refactor... VSCode auto beautifier...

Reply:
@dnviti could you update the pull request or make a new pull request without the auto beautifier? :)